### PR TITLE
Fixes #25757 - remove undefined method disable_taxonomies

### PR DESCRIPTION
**Pull request by rabajaj0509**

### CODE REVIEW

**Review by kgaikwad — APPROVED** [review]
Code wise looks legitimate!

File: test/functional/discovered_hosts_controller_test.rb [hunk 1]
```diff
@@ -84,27 +84,27 @@ def test_edit_form_build_mode_enabled
...
-
-      managed_host = Host.find(host.id)
-      assert managed_host.build
-      assert_redirected_to host_url(managed_host)
-      assert_equal hostgroup.id, managed_host.hostgroup_id
-      assert_match(/Successfully/, flash[:success])
-    end
+    organization = FactoryBot.create(:organization)
+    location = FactoryBot.create(:location)
+    domain = FactoryBot.create(:domain, :organizations => [organization], :locations => [location] )
```
Comment (tbrisker) [inline test/functional/discovered_hosts_controller_test.rb]: You don't need to manually create the taxomonies, the domain factory assigns them by default - https://github.com/theforeman/foreman/blob/develop/test/factories/domain.rb#L9 and so does the hostgroup factory, so if you assign different ones here you may get wrong result

Comment (rabajaj0509) [reply test/functional/discovered_hosts_controller_test.rb]: good catch, thanks! rebased with the corrections.

File: test/functional/discovered_hosts_controller_test.rb [hunk 2]
```diff
@@ -84,27 +84,25 @@ def test_edit_form_build_mode_enabled
...
-        } }, session: set_session_user_default_manager
-
-      managed_host = Host.find(host.id)
-      assert managed_host.build
-      assert_redirected_to host_url(managed_host)
-      assert_equal hostgroup.id, managed_host.hostgroup_id
-      assert_match(/Successfully/, flash[:success])
-    end
+    domain = FactoryBot.create(:domain)
+    hostgroup = FactoryBot.create(:hostgroup, :with_subnet, :with_environment, :with_rootpass, :with_os, :domain => domain, organizations: domain.organizations, locations: domain.locations)
```
Comment (tbrisker) [inline test/functional/discovered_hosts_controller_test.rb]: I'm not sure setting org/loc here is needed as well, the factory default should be the same as the domain one.

**Review by tbrisker — COMMENTED** [review]
Looks like there are 10 other tests failing

File: test/functional/discovered_hosts_controller_test.rb (hunk 2 above)
Comment (rabajaj0509) [reply test/functional/discovered_hosts_controller_test.rb]: Yeah, earlier I had not set the loc/org here but still I was facing same failures. So I just gave this a try.

**Review by tbrisker — APPROVED** [review]
While there are failing tests, this at least fixes this test. merging for now and looking forward to another pr to fix the rest of them.

**Comment by rabajaj0509** [comment]
@tbrisker thanks! I am almost done with the other pr that fixes the rest of the tests :)

**Pull request merged**